File: 02.c
#include <stdio.h>
#include <stdlib.h>
#include <string.h>

#define __ {
#define _ }
#define max(x,y) x>y?x:y

int calc(char* line,int rm,int gm,int bm) __ int r=0,g=0,b=0;
  for(char* p=strchr(line,':')+1; p; p=strchr(p+6,' ')) __
    int n; char color[50];
    sscanf( p, " %d %s", &n, color );
    switch(color[0]) __
      case 'r': r=max(r,n); break;
      case 'g': g=max(g,n); break;
      case 'b': b=max(b,n); break; _ _
  int result = r*g*b;
  if(r<=rm&&g<=gm&&b<=bm) result += atoi(line+5)*100000;
  return result; _

int main() __ int s = 0;
  FILE* fp=fopen("02.dat","rt");
  for(char line[500]; fgets(line,sizeof(line),fp);) s += calc(line,12,13,14);
  fclose(fp);
  printf("%d %d\n",s/100000,s%100000); _

File: 02.py
lines = open("02.dat","rt").readlines()

def calc(line,rm,gm,bm):
  rgb={'r':0,'g':0,'b':0}
  head,tail = line.split(":")
  for cubes in tail.split(";"):
    for cube in cubes.split(","):
      n,color = cube.split()
      rgb[color[0]]=max(rgb[color[0]],int(n))
  power = rgb['r']*rgb['g']*rgb['b']
  if rgb['r']<=rm and rgb['g']<=gm and rgb['b']<=bm:
    power += int(head[5:])*1000000
  return power

print(divmod(sum(calc(line,12,13,14)for line in lines),1000000))
